crates/aranya-daemon/src/sync/task/quic.rs

```diff
@@ -0,0 +1,597 @@
+pub use psk::PskStore;
+pub use version::Version;
+
+/// Errors specific to the QUIC syncer
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// QUIC Connection error
+    #[error("QUIC connection error: {0}")]
+    QuicConnectionError(#[from] s2n_quic::connection::Error),
+    /// QUIC Stream error
+    #[error("QUIC stream error: {0}")]
+    QuicStreamError(#[from] s2n_quic::stream::Error),
+    /// QUIC client config error
+    #[error("QUIC client config error: {0}")]
+    ClientConfig(anyhow::Error),
+    // TODO: Improve message
+    /// Invalid PSK used for syncing
+    #[error("Invalid PSK used when attempting to sync")]
+    InvalidPSK,
+    /// QUIC server config error
+    #[error("QUIC server config error: {0}")]
+    ServerConfig(anyhow::Error),
+}
+
+/// QUIC syncer state used for sending sync requests and processing sync responses
+pub struct State {
+    /// QUIC client to make sync requests and handle sync responses.
+    client: QuicClient,
+    /// Address -> Connection map used for re-using connections
+    /// when making outgoing sync requests
+    conns: BTreeMap<Addr, Connection>,
+    /// Shared PSK store
+    store: Arc<PskStore>,
+}
+
+impl SyncState for State {
+    #[allow(clippy::manual_async_fn)]
+    #[instrument(skip(syncer, sink))]
+    /// Syncs with the peer.
+    /// Aranya client sends a `SyncRequest` to peer then processes the `SyncResponse`.
+    fn sync_impl<S>(
+        syncer: &mut Syncer<Self>,
+        id: GraphId,
+        sink: &mut S,
+        peer: &Addr,
+    ) -> impl Future<Output = SyncResult<()>> + Send
+    where
+        S: Sink<<crate::EN as Engine>::Effect> + Send,
+    {
+        async move {
+            // Sets the active team before starting a QUIC connection
+            syncer.state.store.set_team(id.into_id().into());
+
+            let stream = syncer
+                .connect(peer)
+                .await
+                .inspect_err(|e| error!("Could not create connection: {e}"))?;
+            // TODO: spawn a task for send/recv?
+            let (mut recv, mut send) = stream.split();
+
+            // TODO: Real server address.
+            let server_addr = ();
+            let mut sync_requester = SyncRequester::new(id, &mut Rng, server_addr);
+
+            // send sync request.
+            syncer
+                .send_sync_request(&mut send, &mut sync_requester, peer)
+                .await
+                .map_err(|e| SyncError::SendSyncRequest(Box::new(e)))?;
+
+            // receive sync response.
+            syncer
+                .receive_sync_response(&mut recv, &mut sync_requester, &id, sink, peer)
+                .await
+                .map_err(|e| SyncError::ReceiveSyncResponse(Box::new(e)))?;
+
+            Ok(())
+        }
+    }
+}
+
+impl State {
+    /// Creates a new instance
+    pub fn new(psk_store: Arc<PskStore>) -> SyncResult<Self>
+where {
+        // Create Client Config (INSECURE: Skips server cert verification)
```

Comment:
Don't need to capitalize unless the comment includes a specific type name like `ClientConfig`:
```suggestion
        // Create client config (INSECURE: skips server cert verification)
```